source/fuzz/fuzzer_pass_add_parameters.cpp

```diff
@@ -38,8 +40,23 @@ void FuzzerPassAddParameters::Apply() {
     const auto* type =
         GetIRContext()->get_type_mgr()->GetType(type_inst->result_id());
     assert(type && "Type instruction is not registered in the type manager");
+
     if (TransformationAddParameter::IsParameterTypeSupported(*type)) {
       type_candidates.push_back(type_inst->result_id());
+    } else if (type->kind() == opt::analysis::Type::kPointer) {
+      // Pointer types with storage class Private and Function are allowed.
+      SpvStorageClass storage_class =
+          fuzzerutil::GetStorageClassFromPointerType(GetIRContext(),
+                                                     type_inst->result_id());
+      // Check if the pointee type is supported.
+      auto pointee_type = GetIRContext()->get_type_mgr()->GetType(
+          fuzzerutil::GetPointeeTypeIdFromPointerType(GetIRContext(),
+                                                      type_inst->result_id()));
+      if (TransformationAddParameter::IsParameterTypeSupported(*pointee_type) &&
+          (storage_class == SpvStorageClassPrivate ||
+           storage_class == SpvStorageClassFunction)) {
+        type_candidates.push_back(type_inst->result_id());
+      }
```

Comment:
I think we can move this check into `TransformationAddParameter::IsParameterTypeSupported`.

Reply:
We could also add support for `Workgroup` storage class. It's the same as `Private` storage class but the initializer is `0`. See `TransformationAddGlobalVariable` docs for more details.